### PR TITLE
replace unicode control chars as they can't be parsed by YAMLParser

### PR DESCRIPTION
**Pull request by RamakrishnaChilaka**
The randomized field names may contain Unicode characters. YAMLParser is skipping these control characters during serialization/deserialization that we are doing in `testFromXContent` test case. 

Randomized field name: "æÆ\u0000¢¡Èýñ«Ò"
Parsing fromXContent: "æÆ ¢¡Èýñ«Ò"

Space replaces the control character NUL, leading to the test case failure.

issue-ref: https://github.com/opensearch-project/OpenSearch/issues/6411#issue-1593545979

### CODE REVIEW

**Review by saratvemulapalli — COMMENTED** [review]
Thanks @RamakrishnaChilaka for contributing a change. 

1. Please sign your commit using `git commit -s`for DCO[1]
2. Can you add a test to verify replacing unicodeControlCharacters works (I know its a test for test framework :))
3. Add an entry to `CHANGELOG.md`[2] in unreleased 2.x section (as we will backport this to 2.x) if not @andrross will chase you. 

[1] https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
[2] https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#changelog

**Comment by andrross** [comment]
> 3\. Add an entry to `CHANGELOG.md`[2] in unreleased 2.x section (as we will backport this to 2.x) if not @andrross will chase you.

We can skip the changelog for this test-related change. I've added the label.

**Comment by RamakrishnaChilaka** [comment]
Fixed the gradle check @saratvemulapalli. Please review.

File: server/src/test/java/org/opensearch/search/fetch/subphase/highlight/HighlightFieldTests.java [hunk 1]
```diff
@@ -53,19 +53,32 @@ public class HighlightFieldTests extends OpenSearchTestCase {
 
     public static HighlightField createTestItem() {
         String name = frequently() ? randomAlphaOfLengthBetween(5, 20) : randomRealisticUnicodeOfCodepointLengthBetween(5, 20);
```
Comment (dblock) [inline server/src/test/java/org/opensearch/search/fetch/subphase/highlight/HighlightFieldTests.java:55]: Any reason why we didn't modify `randomAlphaOfLengthBetween` and `randomRealisticUnicodeOfCodepointLengthBetween` to not return unicode control characters?

**Pull request merged**